hawkular-inventory-rest-api/src/main/rest-doc/base.adoc

```diff
@@ -16,3 +16,362 @@ When you send JSON data with a `POST` or `PUT` request, you _must_ add the follo
+E.g.
+
+  PUT /hawkular/inventory/entity/f;My%20Feed/r;RootResource/r;child-resource
+
+  {
+    "properties": {
+      "admin's personal notes": "Seems to have stabilized in the new version..."
+    }
+  }
+
+This will update the resource with ID "child-resource" contained within a "RootResource" resource under a feed with
+ID "My Feed" (under the current tenant specified by the header) to have the above specified properties.
+
+===== Delete
+To delete a resource, send a DELETE request to its `/entity/...` endpoint. Deleting an entity will also recursively
+delete all the child entities contained within it (i.e. deleting a feed will delete all resources, metrics, resource
+types and metric types defined by it). It will NOT delete the entities otherwise associated with it. I.e. if a resource
+incorporates a metric and the resource is deleted, the metric still exists.
+
+===== Retrieving Tree Hash
+The identity hash of entities that support it is based on the Merkle tree hash of their child entities. To retrieve this
+tree hash (for example to quickly check if your local entity tree is the same), you can use:
+
+  /hawkular/inventory/entity/f;feedId/treeHash
+
+==== Traversing Many Entities
+
+Inventory data forms a graph. Using the `/hawkular/inventory/traversal` endpoint one can compose traversals of this
+graph that can search for certain entities. The traversals that can be expressed using the URL do not have the full
+strength of general purpose graph-traversal language like http://tinkerpop.incubator.apache.org/[Gremlin] but many
+things can be achieved with them.
+
+NOTE: In the future, there is a plan to support full Gremlin-based access to inventory.
+
+The URL traversal of the inventory is an extension of the format of canonical path. E.g.
+
+  /hawkular/inventory/traversal/f;feedId/r;resourceId
+
+is a valid URL traversal, because `/f;feedId/r;resourceId` is a valid (tenantless) canonical path.
+
+To be able to formulate efficient traversals, it is good to understand the inventory organization. Please consult
+the http://www.hawkular.org/docs/components/inventory/index.html#inventory-organization[documentation] for its details.
+
+Now to a more formal definition of the URL traversal (for the technically inclined, the traversal is described using an
+http://github.com/hakwular/hawkular-inventory/hawkular-inventory-rest-api/src/main/antlr4/org/hawkular/inventory/rest/HawkularInventoryGetUri.g4[Antlr4 grammar]).
+
+===== Filtering Entities
+
+Each path segment in the URL traversal expresses a filter on the "current" set of entities. The filtering is performed
+within a single URL path segment using a notation similar to HTTP matrix parameters.
+
+Given an example traversal `/hawkular/inventory/traversal/f;feedId;name=My%20Feed/r;resourceId` we'll be talking about
+a single path segment, e.g. `f;feedId;name=My%20Feed` or `r;resourceId`.
+
+Given the example `f;feedId;name=My%20Feed` you can spot two things:
```

Comment:
is it necessary to specify the `feedId` here? The example doesn't make much sense to me. If I know the id and it's unique, what's the point in narrowing the search with specifying the name?